### DISCUSSION TITLE
Sync -> Get Watched update

### DISCUSSION THREAD

**Discussion by michaldrabik** [trakt/trakt-api#397] [Q&A] [answered]
Hello,
I personally feel like `https://api.trakt.tv/sync/watched` could use some versions with more params available.

What I mean is that currently only param is either `movies` or `shows` which can easily lead to fetching a gigantic responses for users with chunky collections. Since currently this endpoint accepts either shows or movies type I believe we should be able to also provide an array of show/movies ids to further limit response size to just whats needed.

Quite often there is a need to just check user's progress for a single show (or just 2,3 of them) and there surely is no need to fetch everything all the time.

**Comment by rectifyer** [comment, marked as answer]
This method is designed to fully sync the entire watched/collection from Trakt to your app. Not plans to add filtering per show since that would make caching way less effective.

If you know the specific shows that need updates, you can use https://trakt.docs.apiary.io/#reference/shows/watched-progress/get-show-watched-progress for a single show.

Just curious how you are determining only a single show needs an update and not more?

**Reply by michaldrabik** [reply]
I actually quite forgot this endpoint existed, thanks.

This is mostly for the case where we would like to export an episode of a single show that user is tracking and we want to make sure that we do not add additional play if episode has been watched already.

**Reply by michaldrabik** [reply]
With a second glance `https://trakt.docs.apiary.io/#reference/shows/watched-progress/get-show-watched-progress` will not do the job probably because as far as I can see it does not include marked not aired episodes.

It is kind of strange actually:

1. I just marked Mandalorian S03 E07 (not yet aired) as watched via website
2. It can be seen in website History
3. It's NOT reflected in website Progress and `https://api.trakt.tv/shows/137178/progress/watched` endpoint
4. It IS reflected in `https://api.trakt.tv/sync/watched/shows`

Is this all as intended?

If yes then it seems that In a case when I'd like to mark S03 E07 play via the app and send it to Trakt BUT skip that sync if play of this episode already exists in user account I'm left with `https://api.trakt.tv/sync/watched/shows`
